pkg/sources/v1alpha2/ping_client.go

```diff
@@ -87,7 +87,15 @@ func (c *pingSourcesClient) DeletePingSource(name string) error {
 }
 
 func (c *pingSourcesClient) GetPingSource(name string) (*v1alpha2.PingSource, error) {
-	return c.client.Get(context.TODO(), name, metav1.GetOptions{})
+	source, err := c.client.Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
```

Comment:
```suggestion
		return nil, knerrors.GetError(err)
```

(import it as `	knerrors "knative.dev/client/pkg/errors"`)

Reply:
we'd need to align catching this error at individual resource-client level across the code, lets begin from here, I will open another issue with details